File: Java/lab_temp/bubble_sort.java
package lab_temp;

public class bubble_sort
{
    public static void main(String[] args)
    {
        int arr[] = {1, 3, 7, 2, 5, 4, 6};
        int n = arr.length;
        
        for(int i=0; i<n; i++)
        {
            System.out.print(arr[i] + " ");
        }

        System.out.println();

        for(int i=0; i<n; i++)
        {
            for(int j=0; j<n-i-1; j++)
            {
                if(arr[j] > arr[j+1])
                {
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
            }
        }

        for(int i=0; i<n; i++)
        {
            System.out.print(arr[i] + " ");
        }
        System.out.println();

    }
    
}

File: Algorithms/Sorting/bubble_sort.cpp
// Bubble Sort is the simplest sorting algorithm that works by
// repeatedly swapping the adjacent elements if they are in wrong order.

#include <bits/stdc++.h>
using namespace std;

void bubble_sort(vector<int> &arr) {
  for (int i = 0; i < arr.size(); i++) {
    for (int j = 0; j < arr.size() - i - 1; i++) {
      if (arr[j] > arr[j + 1]) {
        swap(arr[i], arr[j]);
      }
    }
  }
}

void print_v(vector<int> v) {
  for (auto x : v) {
    cout << x << " ";
  }
  cout << endl;
}

int main(int argc, char const *argv[]) {
  vector<int> arr = {1, -2, 3, 4, 7, 6, 5};
  print_v(arr);

  bubble_sort(arr);

  print_v(arr);

  return 0;
}
